packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -109,15 +124,42 @@ private IReadOnlyList<ParameterProvider> BuildConstructorParameters()
                 {
                     if (!property.IsReadOnly)
                     {
-                        constructorParameters.Add(new ParameterProvider(property)
+                        var parameter = new ParameterProvider(property)
                         {
                             Type = propertyType.InputType
-                        });
+                        };
+                        if (!parameterNames.Contains(parameter.Name))
+                        {
+                            constructorParameters.Add(parameter);
+                        }
                     }
                 }
             }
 
-            return constructorParameters;
+            return (constructorParameters, constructorInitializer);
+
+            static ConstructorSignature? GetBaseConstructor(CSharpType? baseType)
```

Comment:
nit: missing access modifier

Reply:
this is a local method within a scope, therefore no access modifier is allowed on this method.